tests/download/test_client.py

```diff
@@ -89,23 +83,28 @@ def test_hash_collision(download_request: DownloadRequest) -> None:
     # Give all requests same hash
     download_request.get_hashed_name = lambda: "same_hash"
 
-    request2 = copy.deepcopy(download_request)
-    request3 = copy.deepcopy(download_request)
-    request3.post_values = {"zero": 0}
+    structurally_same_request = copy.deepcopy(download_request)
+    structurally_different_request = copy.deepcopy(download_request)
+    structurally_different_request.post_values = {"zero": 0}
 
-    client.download(download_request)
-    client.download(request2)
+    client.download([download_request])
+    client.download([structurally_same_request])
 
     with pytest.raises(HashedNameCollisionException):
-        client.download(request3)
+        client.download([structurally_different_request])
+
+
+def test_check_cached_request_is_matching(download_request: DownloadRequest) -> None:
+    """Checks that when the request is saved (and loaded) the method correctly recognizes it's matching.
+    Ensures no issues with false detections if the jsonification changes values."""
+    client = DownloadClient()
 
-    # Check that there are no issues with re-loading
-    request4 = copy.deepcopy(download_request)
-    request4.post_values = {"transformed-when-saved": [(1, 2)]}
+    request = copy.deepcopy(download_request)
```

Comment:
Do not see why copy  here is necessary, download_request is reconstructed for every test and there is no comparison with original in the test.

Reply:
done